module/signature/errors.go

```diff
@@ -19,6 +19,15 @@ var (
 
 	// ErrInvalidChecksum indicates that the index vector's checksum is invalid
 	ErrInvalidChecksum = errors.New("index vector's checksum is invalid")
+
+	// ErrIdentitySignature indicates that a signature is an 'identity signature', which we
+	// reject as invalid by convention.
+	// Context: If nodes generate their public-private key pairs honestly, there is vanishing
+	// probability of generating identity signatures (individual as well as aggregated signatures).
+	// However, (colluding) byzantine nodes could force the generation of identity signatures, to
+	// undermine cryptographic safety. Therefore, we exclude the identity signature from the set
+	// of accepted signatures.
+	ErrIdentitySignature = errors.New("rejecting identity signature")
```

Comment:
I have the soft preference to _not_ include statements about the logic leading to the error in the description of the error. If you use the error in a different location, the application-specific context usually doesn't carry over. 

For example, you write:
> failed the module safety final verification

But what if we wanted to use the error when checking an aggregated signature that somebody else sent us? For example, just hypothetically consider the scenario
*  `SignatureAggregatorSameMessage.VerifyAggregate` would respond only with error returns
* in this case, returning an `ErrIdentitySignature` would be reasonable from the perspective of `VerifyAggregate`
* but your error description specifically talks about "safety final verification", which `VerifyAggregate` does not do.

I am not sure I am able to explain this properly ... Just trying to say that in the context of the _current_ implementation, the error description makes sense ... but it imposes additional refactoring work (that often is overlooked) in case you want to use the same error in a different context. 

My recommendation is: 
* the error should describe the failure that you reached (identity signature) 
* and avoid describing the logic (safety final verification) that _leads_ to the failure case. As we document the error returns for each method, you can include the comments on the failure path in the method's documentation here: https://github.com/onflow/flow-go/blob/080d4ffbd04a927c7f2fa5178493ca8dcd9c886d/module/signature/aggregation.go#L169-L175